### DISCUSSION TITLE
Multiple Table Insert / Many to Many Using Dart/Flutter In Single Transaction

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#7364] [Questions]
Is there a way to insert into multiple tables at once within Dart/Flutter? If not, is there a link to some docs on how to do it with a function or rPC?

**Comment by GaryAustin1** [comment]
A Postgres function and calling with rpc is the way.  I don't have a great example but Googling showed more possibilities than this one.
https://stackoverflow.com/questions/69595895/how-to-insert-in-multiple-tables-with-a-single-api-call-in-supabase
You basically pass your needed data in the rpc call and then use a Postgres function to make multiple SQL insert calls with that data.  The nice thing is if any fail all rollback, which you can't do with separate inserts to the API.

**Reply by jtkeyva** [reply]
Thanks, I'll give it try! Haven't tried rpc yet

**Reply by jtkeyva** [reply]
Found this too:
https://github.com/supabase/supabase/discussions/3732

**Reply by GaryAustin1** [reply]
That looks like what you are looking for.
